app/uk/gov/hmrc/hecstubs/controllers/AccountingPeriodController.scala

```diff
@@ -0,0 +1,111 @@
+import java.util.UUID
+
+@Singleton
+class AccountingPeriodController @Inject() (cc: ControllerComponents) extends BackendController(cc) with Logging {
+
+  def accountingPeriod(ctutr: String, startDate: String, endDate: String): Action[AnyContent] = Action { request =>
+    val correlationId = request.headers.get("CorrelationId").getOrElse(UUID.randomUUID().toString)
+    val environment   = request.headers.get("Environment").flatMap(Environment.getEnvironmentFromString)
+    environment match {
+      case Some(_) => validatedDataResponse(ctutr, startDate, endDate, correlationId)
+      case None    =>
+        logger.info(
+          s"Call failed to fetch  Company Accounting Period details as Environment in header is invalid"
+        )
+        BadRequest
+    }
+  }
+
+  private def validatedDataResponse(ctutr: String, startDate: String, endDate: String, correlationId: String) =
+    validateInputData(ctutr, startDate, endDate, correlationId) match {
+      case Invalid(errorList) =>
+        val errorResponse = ErrorResponse(errorList.toList)
+        logger.info(
+          s"Responding to call for Company Accounting Period failed with error Responses : ${errorResponse.toString()}"
+        )
+        BadRequest(Json.toJson(errorResponse))
+
+      case Valid(data) =>
+        val responseJson = Json.toJson(data)
+        logger.info(
+          s"Responding to call for Company Accounting Period  for UTR $ctutr , startDate: $startDate and  endDate: $endDate with JSON: ${responseJson.toString()}"
+        )
+        Ok(responseJson)
+    }
+
+  @SuppressWarnings(Array("org.wartremover.warts.Any"))
+  private def validateInputData(
+    ctutr: String,
+    startDate: String,
+    endDate: String,
+    correlationId: String
+  ): ValidatedNel[ErrorResult, CompanyAccountingPeriod] = {
+
+    val message    = "Submission has not passed validation."
+    val ctutrRegex = "^[0-9]{10}$"
+    val dateRegex  =
+      "^(((19|20)([2468][048]|[13579][26]|0[48])|2000)[-]02[-]29|((19|20)[0-9]{2}[-](0[469]|11)[-](0[1-9]|1[0-9]|2[0-9]|30)|(19|20)[0-9]{2}[-](0[13578]|1[02])[-](0[1-9]|[12][0-9]|3[01])|(19|20)[0-9]{2}[-]02[-](0[1-9]|1[0-9]|2[0-8])))$"
+
+    val ctutrValidation: ValidatedNel[ErrorResult, CTUTR] =
+      if (ctutr.matches(ctutrRegex)) Valid(CTUTR(ctutr))
+      else
+        Validated.invalidNel(
+          ErrorResult(InvalidCTUTR, s"$message Invalid parameter ctutr.")
+        )
+
+    val startDateValidation: ValidatedNel[ErrorResult, LocalDate] =
+      if (startDate.matches(dateRegex)) Valid(LocalDate.parse(startDate))
+      else
+        Validated.invalidNel(
+          ErrorResult(InvalidStartDate, s"$message Invalid query parameter startDate.")
+        )
+
+    val endDateValidation: ValidatedNel[ErrorResult, LocalDate] =
+      if (endDate.matches(dateRegex)) Valid(LocalDate.parse(endDate))
+      else
+        Validated.invalidNel(
+          ErrorResult(InvalidEndDate, s"$message Invalid query parameter endDate.")
+        )
```

Comment:
feel free to ignore, possibly could save a few lines of code by turning this into a function and re-using for both from and to dates

Reply:
i thought it myself as well, but then again it's just one line. So i though even if i convert it into function, i have to call that function at all places and again it will be one line . So i dropped the idea. What do you think?